app/src/resources/camera/usePreviewImage.ts

```diff
@@ -1,44 +1,56 @@
-import { useMemo, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 
-import { useCapturePreviewImage } from '@opentrons/react-api-client'
-
-import { MIME_TYPES } from '../dataFiles/constants'
+import {
+  useCapturePreviewImage,
+  useCapturePreviewImageToRun,
+} from '@opentrons/react-api-client'
 
 import type { CameraImageSettings } from '@opentrons/api-client'
 
 export interface UsePreviewImageResult {
   imgPath: string | undefined
   isLoading: boolean
-  takePhoto: () => void
+  photoTaken: () => Promise<void>
 }
 
 export function usePreviewImage(
-  settings: CameraImageSettings
+  settings: CameraImageSettings,
+  runId: string | null
 ): UsePreviewImageResult {
-  const [isLoading, setIsLoading] = useState(false)
-  const { data, refetch } = useCapturePreviewImage(settings, {
-    onSettled: () => {
-      setIsLoading(false)
-    },
-    onError: error => {
-      console.error('Failed to capture preview image', error)
-    },
-  })
-
-  const imgPath = useMemo(() => {
-    if (data == null) return undefined
-
-    const blob = new Blob([data], { type: MIME_TYPES.IMAGE })
-    return URL.createObjectURL(blob)
-  }, [data])
-
-  const takePhoto = (): void => {
-    refetch()
+  const [imageBlob, setImageBlob] = useState<Blob | null>(null)
+
+  const captureImagePreview = useCapturePreviewImage()
+  const captureImagePreviewForRun = useCapturePreviewImageToRun()
+
+  const captureImage =
+    runId == null ? captureImagePreview : captureImagePreviewForRun
+
+  const photoTaken = async (): Promise<void> => {
```

Comment:
```suggestion
  const takePhoto = async (): Promise<void> => {
```
Nit, the verbiage here should be adjusted to better reflect this is a fn and that it produces a side effect.